### PR TITLE
US042 US041 US040 US028 US011 implement state transitions

### PR DESCRIPTION
**Pull request by madeye-matt**
When mandatory events are added, the collection exercise state becomes SCHEDULED.

When the following is satisfied change the state to READY_FOR_REVIEW:
- mandatory collection exercise events have been loaded
- at least one collection instrument has been loaded
- at least one sample file has been loaded

### CODE REVIEW

**Comment by gemmai95** [comment]
[US028: Set and View 'Ready for Live' State of a CE [Large]](https://trello.com/c/X6SD8Tmu/28-us028-set-and-view-ready-for-live-state-of-a-ce-large)

**Comment by jcox-dev** [comment]
[US042: Create and View 'Ready for Review' State of a CE [M]](https://trello.com/c/Fb6CS3es/36-us042-create-and-view-ready-for-review-state-of-a-ce-m)

File: src/main/resources/database/changes/release-10.49.2/us042-deprecate-published.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+UPDATE collectionexercise.collectionexercise SET statefk = 'READY_FOR_LIVE' WHERE statefk = 'PUBLISHED';
```
Comment (benjefferies) [inline src/main/resources/database/changes/release-10.49.2/us042-deprecate-published.sql]: Will this leave surveys which should be 'LIVE' like BRES? in READY_FOR_LIVE?

**Comment by jcox-dev** [comment]
[US041: Create and View 'Scheduled' State of a CE [M]](https://trello.com/c/Quhfml2t/35-us041-create-and-view-scheduled-state-of-a-ce-m)

**Comment by tej99** [comment]
All state transitions work fine from created to ready for live 👍

**Comment by jcox-dev** [comment]
[US040: Create and View 'Created' State of a CE [M]](https://trello.com/c/TI7VB3Jr/34-us040-create-and-view-created-state-of-a-ce-m)

**Pull request merged**